gui/src/2D/xmlLoader.ts

```diff
@@ -0,0 +1,323 @@
+/**
+* Class used to load GUI via XML.
+*/
+
+export class XmlLoader {
+    private _nodes: any = {};
+
+    private _nodeTypes: any = {
+        element: 1,
+        attribute: 2,
+        text: 3
+    };
+
+    private _isLoaded: boolean = false;
+
+    private _objectAttributes: any = {
+        "textHorizontalAlignment": 1,
+        "textVerticalAlignment": 2,
+        "horizontalAlignment": 3,
+        "verticalAlignment": 4,
+        "stretch": 5,
+    };
+
+    private _parentClass: any;
+
+     /**
+     * Create a new xml loader
+     * @param parentClass Sets the class context. Used when the loader is instanced inside a class and not in a global context
+     */
+    constructor(parentClass = null) {
+        if (parentClass) {
+            this._parentClass = parentClass;
+        }
+    }
+
+    private _getChainElement(attributeValue: any, isGlobal: boolean = false): any {
+        let element = window;
+
+        if (this._parentClass && !isGlobal) {
+            element = this._parentClass;
+        }
+        let value = attributeValue;
+        value = value.split(".");
+
+        for (let i = 0; i < value.length; i++) {
+            element = element[value[i]];
+        }
+        return element;
+
+    }
+
+    private _createGuiElement(node: any, parent: any, linkParent: boolean = true): void {
+        try {
+            let className = this._getChainElement("BABYLON.GUI." + node.nodeName, true);

```

Comment:
That will not work if you are using ES6 modules in the client. The classes are not on the global object anymore.
To make it work you will need to update all GUI classes I'am afraid and reference them like here:
https://github.com/BabylonJS/Babylon.js/blob/master/src/Materials/Node/Blocks/addBlock.ts#L66

Then creating one will be easy:
https://github.com/BabylonJS/Babylon.js/blob/master/src/Materials/Node/nodeMaterial.ts#L1174

Reply:
Oh I was not aware. I will update the classes and change the referencing.

Reply:
This should be fine. Updated the GUI classes and changed the way it is accessed in the xmlLoader.